src/main/java/com/example/gruppajava/controller/ParkingEventController.java

```diff
@@ -0,0 +1,81 @@
+package com.example.gruppajava.controller;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.example.gruppajava.entity.ParkingEvent;
+import com.example.gruppajava.repository.CarRepository;
+import com.example.gruppajava.repository.ParkingEventRepository;
+import com.example.gruppajava.repository.UserRepository;
+
+
+@RestController
+@RequestMapping("/api/")
+public class ParkingEventController {
+    @Autowired ParkingEventRepository parkingEventRepository;
+    @Autowired CarRepository carRepository;
+    @Autowired UserRepository userRepository;
+
+    // Get all parking events.
+    @GetMapping("parkingevents")
+    public Iterable<ParkingEvent> getAllParkingEvents() {
+        return parkingEventRepository.findAll();
+    }
+
+    // Create a new parking event.
+    @PostMapping("parkingevents")
+    public ParkingEvent addParkingEvent(@RequestBody Map<String, String> body) {
+        Long userId = Long.parseLong(body.get("userId"));
+        Long carId = Long.parseLong(body.get("carId"));
+        int parkingslotId = Integer.parseInt(body.get("parkingslotId"));
+
+        ParkingEvent parkingEvent = new ParkingEvent();
+        parkingEvent.setUser(userRepository.findById(userId).get());
+        parkingEvent.setCar(carRepository.findById(carId).get());
+        parkingEvent.setSlotId(parkingslotId);
+        parkingEvent.setStartTime(LocalDateTime.now());
+        parkingEvent.setEndTime(LocalDateTime.now().plusMinutes(10));
+        parkingEvent.setActive(true);
+        parkingEvent.setPaid(false);
+        
```

Comment:
I have a feeling most of this logic should go in a service layer or in a constructor for the entity, something about "controller should only do the routing", but I'm unsure...

Reply:
🤷‍♂️

Reply:
I now have a service layer for creating a new parking event ready. Let me now whether to push to this PR or I will create an Issue where I will push fixes to instead :)